cdk/PgStacInfra.ts

```diff
@@ -200,6 +206,41 @@ export class PgStacInfra extends Stack {
             }
           : undefined,
     });
+
+    // STAC Browser Infrastructure
+    const stacBrowserBucket = new s3.Bucket(this, 'Bucket', {
+      accessControl: s3.BucketAccessControl.PRIVATE,
+      removalPolicy: RemovalPolicy.DESTROY,
+      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+      enforceSSL: true,
+    })
+
+    const stacBrowserOrigin = new cloudfront.Distribution(this, 'stacBrowserDistro', {
+      defaultBehavior: { origin: new origins.S3Origin(stacBrowserBucket) },
+    });
+
+    new StacBrowser(this, "stac-browser", {
+      bucketArn: stacBrowserBucket.bucketArn,
+      stacCatalogUrl: stage === "test" ? "https://stac.dit.maap-project.org" : "https://stac.maap-project.org",
```

Comment:
I think you could do this to explicitly set the domain based on inputs to the deployment
```suggestion
      stacCatalogUrl: props.stacApiCustomDomainName,
```

Reply:
I had this at one point which I think would be fine, we would just have to change it from a `string | undefined` to `string` and ensure it's setup in the environment we deploy to beforehand

Reply:
Maybe we could fall back to the output url from the stacapi API gateway?

Reply:
I think we should make the prop a required input and use the same value in both constructs